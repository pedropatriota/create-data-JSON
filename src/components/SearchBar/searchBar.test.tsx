import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SearchBar } from "..";

const setup = () => {
  const utils = render(<SearchBar />);
  return {
    ...utils,
  };
};

afterEach(() => cleanup);

describe("SearchBar", () => {
  it("should render the initial state correctly", async () => {
    setup();

    const input = (await screen.findByRole("textbox")) as HTMLInputElement;
    const searchContainer = screen.getByTestId("search-container");
    const closeIcon = screen.queryByText(/✖/i);

    expect(input).toBeInTheDocument();
    expect(input).not.toHaveFocus();
    expect(input.value).toBe("");
    expect(searchContainer).toHaveStyle("height: 4rem");
    expect(closeIcon).not.toBeInTheDocument();
  });

  it("should render correctly after focus", async () => {
    const { container } = setup();

    const input = (await screen.findByRole("textbox")) as HTMLInputElement;
    const searchContainer = screen.getByTestId("search-container");

    await userEvent.click(input);
    fireEvent.focus(input);

    expect(searchContainer).toHaveStyle("height: 25rem");
    expect(input).toHaveFocus();
  });

  it("should be able type in the input and fetch the API", async () => {
    setup();

    const input = (await screen.findByRole("textbox")) as HTMLInputElement;
    const searchContainer = screen.getByTestId("search-container");

    await userEvent.click(input);
    fireEvent.focus(input);

    expect(searchContainer).toHaveStyle("height: 25rem");
    expect(input).toHaveFocus();

    await userEvent.type(input, "Rick Sanchez");
    expect(input.value).toBe("Rick Sanchez");

    const name = await screen.findAllByText("Rick Sanchez");

    const closeIcon = screen.queryByText(/✖/i);

    expect(closeIcon).toBeInTheDocument();
    expect(name.length).toBe(4);
  });

  it("should collapse the dropdown", async () => {
    setup();

    const input = (await screen.findByRole("textbox")) as HTMLInputElement;
    const searchContainer = screen.getByTestId("search-container");

    await userEvent.click(input);
    fireEvent.focus(input);
    expect(searchContainer).toHaveStyle("height: 25rem");
    await userEvent.type(input, "Rick Sanchez");

    const name = await screen.findAllByText("Rick Sanchez");
    const closeIcon = await screen.findByText(/✖/i);

    await userEvent.click(closeIcon);

    expect(input.value).toBe("");
    expect(searchContainer).toHaveStyle("height: 4rem");
  });
});
