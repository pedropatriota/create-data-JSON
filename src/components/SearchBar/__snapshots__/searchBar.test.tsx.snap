// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshot 1`] = `
<div>
  <div
    class="sc-aXZVg kWhiCM"
    data-testid="search-container"
  >
    <div
      class="sc-gEvEer cxURqX"
    >
      <input
        class="sc-eqUAAy kEUaV"
        placeholder="Make your search..."
        value=""
      />
    </div>
    <ul
      class="sc-dcJsrY fAlmUc"
    />
  </div>
</div>
`;
